test/language/module-code/source-phase-import/import-source-binding-name.js

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2024 Chengzhong Wu. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: >
+  ImportBinding in ImportDeclaration may be 'source'
+esid: sec-modules
+info: |
+  ImportDeclaration:
+    import source ImportedBinding FromClause ;
+
+negative:
+  phase: resolution
+  type: SyntaxError
+features: [source-phase-imports]
+flags: [module]
+---*/
+
+$DONOTEVALUATE();
+
+import source source from '';
```

Comment:
There is no requirement that an empty specifier will fail resolving. Can we use something `from '<do not resolve>'`, and explicitly document it in https://github.com/tc39/test262/blob/main/INTERPRETING.md?

Reply:
Updated with two special specifier:

- `<do not resolve>`: which should be resolved with a TypeError throw completion.
- `<abstract module source>`: which should be resolved with an instance of %AbstractModuleSource%`, as there is not alternative ModuleSource in the ecma262 at the moment.